k8s/upgrade/src/bin/upgrade-job/helm/client.rs

```diff
@@ -201,8 +207,11 @@ impl HelmReleaseClient {
             .with_namespace(self.namespace.as_str())
             .build()
             .await?;
-        // Ref: https://helm.sh/docs/chart_best_practices/custom_resource_definitions
-        install_missing_crds(k8s_client.crd_api(), chart_dir.join("crds")).await?;
+
+        if install_crds {
```

Comment:
oh seems like a fix? probably worth separate commit

Reply:
This is a part for the refactor. The upgrade dry-run implementation doesn't do this step, this step needs to avoided when running the dry-run through the core upgrade method.